### DISCUSSION TITLE
Support for Spring-data-mongodb 4.x.x

### DISCUSSION THREAD

**Discussion by ironmetvm** [flamingock/mongock#20] [General]
While Upgrading spring-data-mongodb from 3.x.x to 4.x.x , I am getting issues with com.github.cloudyrock.mongock's mongodb-springdata-v3-driver . When i thought of updating it to v4  didn't find the suitable jar . Need help to continue using mongock for 4.x.x version of spring-data-mongodb.

**Comment by dieppa** [comment]
Hello @ironmetvm, take a look to [this issue](https://github.com/mongock/mongock/issues/613), I believe you will find the solution there.

**Reply by ironmetvm** [reply]
Hi @dieppa , Thanks for the response.
I migrated to io.mongock , but facing issue with triggering 'buildInitializingBeanRunner' on demand . As I want to run the migration whenever 1st request comes for a tenant. 
Below the earlier code which was running fine , but with the upgrade to io.mongock (5.3.3) there is no execute() for **buildInitializingBeanRunner**.

` MongockSpring5.builder()
                .setDriver(driver)
                .addChangeLogsScanPackage("com.package.path")
                .dontFailIfCannotAcquireLock()
                .setSpringContext(appContext)
                .setEnabled(runMigrations)
                .setLegacyMigration()
                .buildInitializingBeanRunner()
                .execute();`

Am I missing out something here with io.mongock ? Need your help to resolve this.

**Reply by ironmetvm** [reply]
Hi @dieppa , kindly ignore the above issue , we got the solution & working on it.

**Reply by dieppa** [reply]
I am glad you found a way. Can you please  post it for other users can learn?
Thanka
